lib/consts/consts.go

```diff
@@ -1,12 +1,32 @@
 package consts
 
 import (
+	"fmt"
+	"runtime"
+	"runtime/debug"
 	"strings"
 )
 
 // Version contains the current semantic version of k6.
-//nolint:gochecknoglobals
-var Version = "0.26.0-dev"
+var Version = "0.26.0-dev" //nolint:gochecknoglobals
+
+// VersionDetails can be set externally as part of the build process
+var VersionDetails = "" // nolint:gochecknoglobals
+
+// GetFullVersion returns the maximully full version and build information for
+// the currently running k6 executable.
+func GetFullVersion() string {
+	goVersionArch := fmt.Sprintf("%s, %s/%s", runtime.Version(), runtime.GOOS, runtime.GOARCH)
+	if VersionDetails != "" {
+		return fmt.Sprintf("%s (%s, %s)", Version, VersionDetails, goVersionArch)
+	}
+
+	if buildInfo, ok := debug.ReadBuildInfo(); ok {
+		return fmt.Sprintf("%s (%s, %s)", Version, buildInfo.Main.Version, goVersionArch)
+	}
+
+	return fmt.Sprintf("%s (dev build, %s)", Version, goVersionArch)
```

Comment:
What do you think about returning information from Git here?

I.e. read the HEAD SHA from the repo (or leave it out if not found) with something like [go-git](https://github.com/src-d/go-git) or even a crude `exec.Cmd()`. That way we'll always have the commit reference, even on public or user-made builds.

This could be extended to also read the version from the nearest tag, to avoid having to update `consts.go` on every release. Though a static fallback would probably still be required for building from source only.

Reply:
If I am understanding you correctly you want to get the version at runtime through some means ... from the current directory ?
This ... doesn't work (or at least doesn't work as we want it).
1. we wouldn't get any version .. in for example docker as there will be no `.git` to read from ;)
2. if someone changes the currently checkout code or is .. in another git repository it's going to read another version ...

If you mean reading it at compile time ... there is no "macros-like" things in golang ... so we can't :) 
We technically (not really) could use [go:generate](https://blog.golang.org/generate) that will update the `consts.go` ... but this will mean that we will have to commit it ... every time we commit ... which also doesn't work (primarily because add the version to the code changes the commit ;) )

Reply:
> you want to get the version at runtime through some means

No, I meant at build time, so yeah, via `go:generate`, though it could update a file excluded from Git (`lib/consts/version_gen.go`?), which could be optionally read at runtime in `consts.go` (e.g. `if gitVersion != "" {...}`).

Not sure if this would work for all scenarios, but it would be useful to have the commit ref whenever possible. Anyway, just a thought.

Reply:
`go:generate` aren't ran automatically you need to call `go generate <path>` to run them ...
Which means that while it would be useful for local development to do `go generate ./... && go install .` , if you forget to `go generate ./...` between checkouts (for example) it's going to tell you the wrong version ... which IMO is worse than not telling you any version :)

Reply:
OK, then I guess injecting it manually with `go build` is the way to go.

I still think an automated solution that stores Git information if available would be helpful for debugging user builds.

Reply:
@imiric, see [this comment](https://github.com/loadimpact/k6/issues/1070#issuecomment-552507335) that I referenced in the PR description. Once we switch to Go modules, `debug.ReadBuildInfo()` _should_ automagically give us something approximating the git commit ID, without us doing any build-time external data injection.